### PR TITLE
[react] Add `value` attribute to `data` element (reopen #36771)

### PR DESCRIPTION
**Pull request by dosentmatter**
The last PR, #36771, was automatically closed by the typescript-bot. It was owner approved by @ferdaber and it had no other responses from reviewers. What else is required to get this merged?

`<data>` should include `value` attribute in addition to the global
attributes:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/data#Attributes

Also add `HTMLDataElement` to global.d.ts

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/data#Attributes
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by ferdaber — APPROVED** [review]
I think the issue is that your changes are unrelated to the test failures caused in the bot. You may have to fix them in order to move forward or manually tag a maintainer to get it merged.

**Comment by jessetrinity** [comment]
Sorry you were blocked by this build issue - we are looking into it. Looks like we managed to get this one a green run though.

**Pull request merged**
